core/services/fluxmonitor/flux_monitor.go

```diff
@@ -878,3 +884,44 @@ func OutsideDeviation(curAnswer, nextAnswer decimal.Decimal, threshold float64)
 	logger.Infow("Deviation threshold met", loggerFields...)
 	return true
 }
+
+const pathologicalSkew = 30 * time.Second
+
+func MakeNewRoundIdleTimer(log *contracts.LogNewRound, idleThreshold models.Duration, clock Clock) <-chan time.Time {
+	timeNow := clock.Now()
+	var roundStarted time.Time
+	if log.StartedAt.IsInt64() {
+		roundStarted = time.Unix(log.StartedAt.Int64(), 0)
+	} else {
+		logger.Errorf("value for log.StartedAt %s would overflow int64, ignoring system clock and setting idle timer to max duration of %s", log.StartedAt.String(), idleThreshold.Duration())
+		return clock.After(idleThreshold.Duration())
+	}
+	if roundStarted.After(timeNow) {
+		logger.Warnf("round started time of %s is later than current system time of %s, ignoring system clock and setting idle timer to max duration of %s. This machine probably has a slow clock", roundStarted.String(), timeNow.String(), idleThreshold.Duration().String())
+		return clock.After(idleThreshold.Duration())
+	}
+	if timeNow.Sub(roundStarted) > pathologicalSkew {
+		logger.Warnf("current system time %s is much later than block timestamp of %s. Either you are processing an old round, or this machine has a fast clock", timeNow.String(), roundStarted.String())
+	}
+	// duration from now until idle threshold = log timestamp + idle threshold - current time
+	timeNow = clock.Now()
+	durationUntilIdleThreshold := roundStarted.Add(idleThreshold.Duration()).Sub(timeNow)
+	if durationUntilIdleThreshold < 0 {
+		logger.Warnf("idle threshold already passed, current time is %s and idle timer expired at %s", timeNow, roundStarted.Add(idleThreshold.Duration()).String())
+		expired := make(chan time.Time, 1)
+		expired <- roundStarted.Add(idleThreshold.Duration())
+		return expired
```

Comment:
This is a tricky case to handle, but right now I'm leaning towards what you did on line 901.  It would allow us to eliminate the `select` you added in `respondToNewRoundLog`, because that could never really happen.  Then, worst case, the current round would time out.

Reply:
Tricky. If we assume the system clock is reliable, the correct thing to do is time out immediately (different behaviour from line 901).

If we assume the system clock is completely bogus, we cannot do anything other than simply waiting idleThreshold seconds from right now.

Which should we do?